### ISSUE TITLE
Missing typescript declaration file ~4.0.0

### ISSUE THREAD

**Issue by JensDebergh** [react-native-community/react-native-device-info#814]
# Bug

<img width="1591" alt="Screenshot 2019-09-25 at 10 50 46" src="https://user-images.githubusercontent.com/6651506/65585249-76992000-df82-11e9-9091-b6f76452e78d.png">

Library version: x.x.x

## Steps To Reproduce

1. Use typescript and install version ~4.0.0
2. Start react native server
...

Describe what you expected to happen:

1. `import DeviceInfo from 'react-native-device-info'` should not give typescript errors

Error:
```
Could not find a declaration file for module 'react-native-device-info'. react-native-device-info/lib/commonjs/index.js' implicitly has an 'any' type.
  Try `npm install @types/react-native-device-info` if it exists or add a new declaration (.d.ts) file containing `declare module 'react-native-device-info';`
```

**Comment by mikehardy** [comment]
@Johan-dutoit this is strange - is there something obvious here?

For reference I'm using this in a typescript project and I don't see a problem. I do this:

```typescript
import DeviceInfo from 'react-native-device-info';
```

then a bunch of this:

```typescript

  // This should be a fast way to get essential info during app boot
  queryEssentialDeviceInfo(): EssentialDeviceInfo | null {
    try {
      console.log('DeviceInfoService::queryEssentialDeviceInfo - fetching info');
      const uniqueId = DeviceInfo.getUniqueIdSync();
      const model = DeviceInfo.getModelSync();
      const manufacturer = DeviceInfo.getManufacturerSync();
      const systemVersion = DeviceInfo.getSystemVersionSync();
      const appVersion = DeviceInfo.getVersionSync();
```

and it all works?

**Comment by mikehardy** [comment]
Oh wait! :man_facepalming:  - I am not representative usage for a module I maintain haha. So maybe there is a problem.

```json
    "react-native-device-info": "3.1.4",
```

**Comment by mikehardy** [comment]
Note that 4.x.x isn't officially published yet: https://www.npmjs.com/package/react-native-device-info

**Comment by JensDebergh** [comment]
Sorry about that tried to use 4.0.0, but somethings not right yet. I kept getting issues related to types not being found.

I guess it's because `index.d.ts` is missing in the root of the project?

**Comment by mikehardy** [comment]
@JensDebergh it's not published for a reason yet - @Johan-dutoit is still working on it

**Comment by Johan-dutoit** [comment]
Having a quick look, the issue seems to be the `types` value in `package.json`

@JensDebergh If you pull down `4.0.0` again and change the following in RNDevice info's `package.json`:

from:
`"types": "lib/typescript/src/index.d.ts",`
to:
`"types": "lib/typescript/index.d.ts",`

Then it should work (I don't have the time to test it right now, but can do so tomorrow).

ps. You might have to restart your editor to pick up the changes
pps.  I'm reopening this to keep a reminder.

**Comment by Johan-dutoit** [comment]
This is fixed in the latest 4.0.x pre-release (can install via `yarn add react-native-device-info@next`).
